service/controller/v24/resource/cloudformation/delete.go

```diff
@@ -100,6 +106,57 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 	return nil
 }
 
+func (r *Resource) disableMasterTerminationProtection(ctx context.Context, masterInstanceName string) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "disabling master instance termination protection")
+
+	i := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("tag:Name"),
+				Values: []*string{
+					aws.String(masterInstanceName),
+				},
+			},
+		},
+	}
+	o, err := cc.AWSClient.EC2.DescribeInstances(i)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, reservation := range o.Reservations {
+
+		if len(reservation.Instances) < 1 {
+			return microerror.Mask(notMasterInstanceExistsError)
+		} else if len(reservation.Instances) > 1 {
+			return microerror.Mask(moreThanOneMasterInstanceExistsError)
+		}
```

Comment:
I am not sure if there is a nicer solution

Reply:
We can do it just like here: https://github.com/giantswarm/aws-operator/blob/master/service/routetable/route_table.go#L91-L93.